module/os/linux/zfs/zpl_file.c

```diff
@@ -1018,6 +1021,9 @@ const struct address_space_operations zpl_address_space_operations = {
 	.writepage	= zpl_writepage,
 	.writepages	= zpl_writepages,
 	.direct_IO	= zpl_direct_IO,
+#ifdef HAVE_VFS_SET_PAGE_DIRTY_BUFFERS
+	.set_page_dirty = __set_page_dirty_buffers,
```

Comment:
```suggestion
	.set_page_dirty = __set_page_dirty_nobuffers,
```

Since we're only wiring this up for the ZFS filesystems we can use `__set_page_dirty_nobuffers()` since there will never be a `buffer_head` attached to the page.  One additional advantage of this is we can update the `configure` check to detect `__set_page_dirty_nobuffers()` which was added in this same upstream patch set which made wiring up `.set_page_dirty` aops mandatory.  This way we only change the behavior for 5.14 and newer kernels.

I've locally tested using `__set_page_dirty_nobuffers()` with the ZTS mmap tests and things work as expected.

Reply:
Thanks, I missed that detail - sounds good to me and I'll modify & re-push the change with this modification, and also will include the commit that introduced the breakage in an amended commit message.

Reply:
Ok @behlendorf I have modified my patch and re-applied with your recommendations - take a look and let me know how you like this version

Reply:
BTW this version is running on my system now, and working fine